src/App.tsx

```diff
@@ -1,24 +1,254 @@
+          .map((todo: Todo) => deleteTodo(USER_ID, todo)),
+      );
+      await getAllTodos();
+    } catch {
+      addErrorMessage(ErrorMessages.DELETE_COMPLETED);
+    } finally {
+      setTodosWithLoader(currentTodos => (
+        currentTodos.filter(todo => todo.completed !== true)
+      ));
+    }
+  }, [completedTodos]);
+
+  const changeStatus = useCallback(async (todoChangeStatus: Todo) => {
+    clearMessage();
+
+    const newTodoStatus = !todoChangeStatus.completed;
+    const todoIdChangeStatus = todoChangeStatus.id;
+
+    setTodosWithLoader(currentTodos => [...currentTodos, todoChangeStatus]);
+
+    try {
+      await updateStatusTodo(todoIdChangeStatus, newTodoStatus);
+      await getAllTodos();
+    } catch {
+      addErrorMessage(ErrorMessages.UPDATE);
+    } finally {
+      setTodosWithLoader(currentTodos => (
+        currentTodos.filter(t => t.id !== todoChangeStatus.id)
+      ));
+    }
+  }, [todos]);
+
+  const changeTitle = useCallback(async (
+    todoChangeTitle: Todo,
+    newTitle: string,
+  ) => {
+    clearMessage();
+    const todoIdChangeTitle = todoChangeTitle.id;
+
+    setTodosWithLoader(currentTodos => [...currentTodos, todoChangeTitle]);
+
+    try {
+      await updateTitleTodo(todoIdChangeTitle, newTitle);
+      await getAllTodos();
+    } catch {
+      addErrorMessage(ErrorMessages.UPDATE);
+    } finally {
+      setTodosWithLoader(currentTodos => (
+        currentTodos.filter(t => t.id !== todoChangeTitle.id)
+      ));
+    }
+  }, [todos]);
+
+  const toggleStatusForAllTodos = useCallback(async () => {
+    clearMessage();
+
+    const statusForUpdate = todos.length !== completedTodos.length;
+
+    const arrayForUpdate = todos.filter(todo => (
+      todo.completed !== statusForUpdate
+    ));
+
+    try {
+      await Promise.all(
+        arrayForUpdate
+          .map((todo: Todo) => updateStatusTodo(todo.id, statusForUpdate)),
+      );
+      await getAllTodos();
+    } catch {
+      addErrorMessage(ErrorMessages.UPDATE_ALL);
+    }
+  }, [todos]);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        {!isLoadingFailed && (
+          <>
+            <Header
+              addNewTodo={addNewTodo}
+              addErrorMessage={addErrorMessage}
+              isSubmitButtonDisabled={isSubmitButtonDisabled}
+              toggleStatusForAllTodos={toggleStatusForAllTodos}
+              isToogleButtonVisible={isToogleButtonVisible}
+            />
```

Comment:
You use `await getAllTodos()` where you set the value for `isLoadingFailed ` in `toggleStatusForAllTodos `, `changeStatus` and other functions, so maybe if there is an error after changing the todos you will not see all previously added todos at all. Maybe it's better to use check with todos length here.